junit/src/main/java/cucumber/api/junit/CucumberRule.java

```diff
@@ -0,0 +1,139 @@
+package cucumber.api.junit;
+
+import cucumber.runtime.ClassFinder;
+import cucumber.runtime.Runtime;
+import cucumber.runtime.RuntimeOptions;
+import cucumber.runtime.RuntimeOptionsFactory;
+import cucumber.runtime.io.MultiLoader;
+import cucumber.runtime.io.ResourceLoader;
+import cucumber.runtime.io.ResourceLoaderClassFinder;
+import cucumber.runtime.junit.FeatureRunner;
+import cucumber.runtime.junit.JUnitReporter;
+import cucumber.runtime.model.CucumberFeature;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.BlockJUnit4ClassRunner;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.InitializationError;
+import org.junit.runners.model.Statement;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+
+public class CucumberRule implements TestRule {
+    private final RuntimeOptions runtimeOptions;
+
+    public CucumberRule() {
+        this(new RuntimeOptionsFactory(CucumberRule.class /* no conf */).create());
```

Comment:
Not sure what you mean. Goal of the rule was to comibe cucumber with other rules (container, ftp/sft/... server, mysql, nosql databaseq etc...)